android/src/ch/opengis/qfield/QFieldOpenExternallyActivity.java

```diff
@@ -41,21 +42,26 @@ protected void onCreate(Bundle savedInstanceState){
         //copy file to a temporary file
         try{
             copyFile( file, cacheFile );
+
+            Uri contentUri =  Build.VERSION.SDK_INT < 24 ? Uri.fromFile(file) : FileProvider.getUriForFile( this, BuildConfig.APPLICATION_ID+".fileprovider", cacheFile );
+
+            Log.d(TAG, "content URI: " + contentUri);
+            Log.d(TAG, "call ACTION_VIEW intent");
+            Intent intent = new Intent();
+            intent.setAction(Intent.ACTION_VIEW);
+            intent.addFlags(Intent.FLAG_GRANT_READ_URI_PERMISSION);
+            intent.setDataAndType(contentUri, mimeType);
+            try{
+                startActivityForResult(intent, 102);
+            }catch (Exception e) {
+                Log.d(TAG, e.getMessage());
+                Toast.makeText( this, "No handler for this type of file.", Toast.LENGTH_LONG).show();
```

Comment:
Can we return an error and expose the message from QML?
For translatability and a consistent styling.

Reply:
I have to check if we can use the java errors as return values...

Reply:
We could at least get back the java error message by handleActivityResults...

Reply:
Can we have a code instead of an error message (--> Translation)

Reply:
I don't know. I'll check on Monday. At least the message or a self defined code...

Reply:
If you end up with the string one, there is this: https://doc.qt.io/qt-5/qtglobal.html#QT_TRANSLATE_NOOP